lms/djangoapps/courseware/grades.py

```diff
@@ -30,6 +31,48 @@
 log = logging.getLogger("edx.courseware")
 
 
+EXCLUDED_LOCATION_CATEGORIES = {'video', 'html', 'discussion'}
+def descriptor_filter(descriptor):
+    """
+    Pass a location filter to the field data cache so that we don't
+    instantiate video descriptors during grading
+    """
+    return descriptor.location.category not in EXCLUDED_LOCATION_CATEGORIES
+
+
+class MaxScoresCache(object):
+    """docstring for MaxScoresCache"""
+    def __init__(self, locations, cache=cache):
+        self.locations = locations
+        self._max_scores_cache = {}
+        self._max_scores_updates = {}
+
+    def fetch_from_cache(self):
+        self._max_scores_cache = cache.get_many(
+            [unicode(loc) for loc in self.locations]
+        )
+
+    def push_to_cache(self):
+        if self._max_scores_updates:
+            cache.set_many(self._max_scores_updates, 180)  # 3 mins
+
+    def add_max_score(self, location, max_score):
+        print "added to cache"
+        loc_str = unicode(location)
+        if self._max_scores_cache.get(loc_str) != max_score:
+            self._max_scores_updates[loc_str] = max_score
+
+    def get_max_score(self, location):
+        loc_str = unicode(location)
+        max_score = self._max_scores_updates.get(loc_str)
+        if max_score is None:
+            max_score = self._max_scores_cache.get(loc_str)
+
+        if max_score is None:
+            max_score = cache.get(loc_str)
```

Comment:
Won't this always return `None`, if it hasn't been preloaded? I'm worried we're making an unnecessary extra network call here.